koku/providers/azure/provider.py

```diff
@@ -99,9 +106,15 @@ def cost_usage_source_is_reachable(self, credential_name,
             storage_accounts = azure_client.storage_client.storage_accounts
             storage_account = storage_accounts.get_properties(resource_group,
                                                               storage_account)
+            azure_service = AzureService(subscription_id, tenant_id, client_id,
+                                         client_secret, resource_group, storage_account)
         except (AdalError, AzureException, ClientException, TypeError) as exc:
```

Comment:
```suggestion
         except (AdalError, AzureException, AzureServiceError, ClientException, TypeError) as exc:
```
`AzureService` raises `AzureServiceError`. If you're adding to this `try` block, then we need to catch that exception. You will need to import `AzureServiceError` from `azure_service`.

Reply:
good _catch_.  I've added it. Resolve this conversation if you think it's done.

Reply:
See: https://azure.github.io/azure-sdk-for-python/exceptions.html and https://github.com/Azure/msrest-for-python/blob/master/msrest/exceptions.py and https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-common/azure/common/exceptions.py

It should be sufficient to catch `ClientException` and `CloudError`, which are the base exception classes for almost everything Azure throws.

Reply:
@maskarb Added

Reply:
@blentz just saw your comment, how do you feel about leaving the redundant catch to be more explicit?